src/game/behaviors/sparkle_spawn_star.inc.c

```diff
@@ -36,11 +36,15 @@ void set_sparkle_spawn_star_hitbox(void) {
     }
 }
 
-void set_home_to_mario(void) {
+void spawned_star_set_target_above_mario(void) {
     vec3f_copy_y_off(&o->oHomeVec, &gMarioObject->oPosVec, 250.0f);
 
-    o->oPosY = o->oHomeY;
+    // Check that the star isn't clipping inside the ceiling
+    if (o->oHomeY > ((gMarioState->ceilHeight - 50))) {
```

Comment:
```suggestion
    if (o->oHomeY > (gMarioState->ceilHeight - 50)) {
```
formatting changes :)